ci/ReleaseTool/PrepCommand.cs

```diff
@@ -118,38 +101,17 @@ public int Run()
                         gitClient.CheckoutRemoteBranch(options.SourceBranch);
 
                         // 3. Makes repo-specific changes for prepping the release (e.g. updating version files, formatting the CHANGELOG).
-                        switch (options.GitRepoName)
+                        if (Common.UpdateVersionFilesWithEngine(gitClient, gitRepoName, options.Version, options.EngineVersions, Logger, "-rc"))
```

Comment:
This now only and commits and pushes if there are actual changes to the version files (this is now consistent with the prepFull commond)

Reply:
Actually, there's a slight edge-case here. Since the Prep command creates PRs from `x.y.z-rc` into `release`, GitHub does not allow creating PRs which do not result in any merge action (e.g. the branches are identical). If we do not push a commit to the rc branches, it may make the rc branch identical to the branch it's meant to merge into. Note: this does not really affect real runs of the release pipeline, since we only release if we have some new things, but dry-runs create a `dry-run/release` branch which is identical to master. Two possibilites: create a commit, even if it's empty. Or change the dry-run behaviour I guess.

I kind of like the commit as a marker for when the rc-creation happened.

Reply:
Hm, but maybe we are fine, as the version file changes should always happen, unless we are re-running the pipeline.